### PR TITLE
Fix #1543.

### LINKED ISSUE #1543: segfault: pre-condition + out parameter
```

module main;
import std.stdio;
class Bar
{
 
    void failMe(out string some)
    in
    {
        assert(true);
    }
    body
    {
        writeln("hello");
    }
}

void main()
{
}

```

```
 ldmd2 -run ldc-bug2.d
0  ldc2                     0x000000010b741d2b llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 43
1  ldc2                     0x000000010b741126 llvm::sys::RunSignalHandlers() + 70
2  ldc2                     0x000000010b7424be SignalHandler(int) + 478
3  libsystem_platform.dylib 0x00007fff8de0252a _sigtramp + 26
Error: Error executing /Volumes/data/code/d/ldc2-1.0.0-beta2-osx-x86_64/bin/ldc2: Segmentation fault: 11
```

### PR DESCRIPTION
**Pull request by JohanEngelen**
It appears that syntaxCopy can be used with the current frontend.

### CODE REVIEW

**Comment by JohanEngelen** [comment]
(to me, this is just one more item on the list of bad things due to lowering in the front-end / AST not representing the source)

File: tests/codegen/func_contracts_gh1543.d [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        some = 123;
+    }
+}
+
+// Bar.failMe codegen order = function, in-contract __require function, out-contract __ensure function
+
+// CHECK-LABEL: define {{.*}} @{{.*}}Bar6failMe
+// CHECK-SAME: i32* dereferenceable(4) %some
+// CHECK: store i32 0, i32* %some
+// CHECK: call void @{{.*}}Bar6failMeMFJiZ9__require
```
Comment (kinke) [inline tests/codegen/func_contracts_gh1543.d]: `call{{.*}} void ...` as IIRC it's `call x86_stdcallcc ...` for 32-bit MSVC. Thanks for the fix!

**Comment by JohanEngelen** [comment]
Fixed the win32 test.

File: tests/codegen/func_contracts_gh1543.d (hunk 1 above)
Comment (JohanEngelen) [inline tests/codegen/func_contracts_gh1543.d]: I knew I would miss one win32-related "{{.*}}" :)

**Pull request merged**